src/com/jsonnetplugin/JsonnetIdentifierReference.java

```diff
@@ -43,19 +76,33 @@ public JsonnetIdentifierReference(@NotNull PsiElement element, TextRange textRan
                 }
             }else if (element instanceof JsonnetObjinside) {
 
-                List<JsonnetObjlocal> locals = new ArrayList<>(((JsonnetObjinside)element).getObjlocalList());
-                for (JsonnetMember m: ((JsonnetObjinside)element).getMembers().getMemberList()){
-                    if (m.getObjlocal() != null){
-                        locals.add(m.getObjlocal());
+                List<JsonnetObjlocal> locals = new ArrayList<>(((JsonnetObjinside) element).getObjlocalList());
+                JsonnetMembers members = ((JsonnetObjinside) element).getMembers();
+                if (members != null) {
+                    for (JsonnetMember m : members.getMemberList()) {
+                        if (m.getObjlocal() != null) {
+                            locals.add(m.getObjlocal());
+                        }
                     }
                 }
-                for (JsonnetObjlocal local: locals) {
+                for (JsonnetObjlocal local : locals) {
                     JsonnetBind b = local.getBind();
                     if (identifier.equals(findIdentifierFromBind(b))) {
                         results.add(new PsiElementResolveResult(b));
                         return results.toArray(new ResolveResult[results.size()]);
                     }
                 }
+
+                JsonnetCompspec comp = ((JsonnetObjinside) element).getCompspec();
+                JsonnetForspec forspec = ((JsonnetObjinside) element).getForspec();
+                if (findComprehensionBinding(results, identifier, comp, forspec))
+                    return results.toArray(new ResolveResult[results.size()]);
+
+            }else if (element instanceof JsonnetArrcomp){
+                JsonnetCompspec comp = ((JsonnetArrcomp) element).getCompspec();
+                JsonnetForspec forspec = ((JsonnetArrcomp) element).getForspec();
+                if (findComprehensionBinding(results, identifier, comp, forspec))
+                    return results.toArray(new ResolveResult[results.size()]);
```

Comment:
This and the below chunk are an unrelated fix, to allow jumping to bindings from the `for x in foo` in list and dictionary comprehensions